hack/docs/configuration_gen_docs.go

```diff
@@ -24,6 +24,10 @@ import (
 	"github.com/aws/karpenter/pkg/utils/options"
 )
 
+var docsLinks = map[string]string{
+	"https://karpenter.sh/unspecified/aws/provisioning/#pod-eni-security-groups-for-pods": "[Pod ENI documentation](../../aws/provisioning/#pod-eni-security-groups-for-pods)",
```

Comment:
Should unspecified be `preview`? or just `docs`?

Reply:
"unspecified" is the default value if no tag is discovered (https://github.com/aws/karpenter/blob/main/pkg/utils/project/project.go#L25).  

But if we change the link in `options.go` to be `karpenter.sh/docs`, then we will replace `karpenter.sh/docs` instead.